lib/Config.js

```diff
@@ -191,10 +191,32 @@ function hdClientLocationConstraintAssert(configHd) {
     return hdclientFields;
 }
 
+function dmfLocationConstraintAssert(location, locationObj) {
+    const {
+        username,
+        password,
+        nsId,
+        repoId,
+        endpoint,
+    } = locationObj.details;
+    if (locationObj.isCold) {
+        assert(typeof username === 'string',
+            'bad config: username must be a string');
+        assert(typeof password === 'string',
+            'bad config: password must be a string');
+        assert(typeof nsId === 'string',
+            'bad config: nsId must be a string');
+        assert(typeof repoId === 'object',
+            'bad config: repoId must be an array');
```

Comment:
```suggestion
        assert.strictEqual(Array.isArray(repoId));
```

Reply:
Thanks, I cherry-picked your commit in https://github.com/scality/cloudserver/pull/4532/commits/89d030bfe9f3ca2d4d1b0e4f340d55f49c486131
It should address most of your comments!